python/ray/workflow/step_executor.py

```diff
@@ -372,7 +397,18 @@ def _workflow_step_executor(
             # Execute sub-workflow. Pass down "outer_most_step_id".
             with workflow_context.fork_workflow_step_context(
                     outer_most_step_id=outer_most_step_id):
-                result = execute_workflow(persisted_output)
+                sub_workflow = persisted_output
+                if resources_equal(runtime_options.ray_options,
```

Comment:
I'm not sure it's valid here. At least for memory usage, it's dynamic and might not be qualified to run.

For example, first one, run and put 10gb into memory and the second one might not be ok to run.